packages/hub/indexing/document-context.js

```diff
@@ -107,6 +108,203 @@ module.exports = class DocumentContext {
     return resource;
   }
 
+  // This takes an arbitrary JSON:API document and makes it safe to
+  // show within the given context. If it can't be shown at all
+  // (because the primary resource is not readable), we return
+  // undefined.
+  //
+  // Otherwise, we go through each included resource and field to
+  // ensure there are valid grants for them. We return a new sanitized
+  // document that strips out any included resources or fields that
+  // were lacking grants.
+  async applyReadAuthorization(document, context={}) {
```

Comment:
@balinterdi @ef4 why does this operate on an arbitrary document as opposed to the document contained in this DocumentContext instance? I'm concerned about sharing the DocumentContext internals outside of its context. Maybe we should be calling `applyReadAuthorization()` without passing a document to it, so that `applyReadAuthorization()` operates on its `this.upstreamDoc`.

Reply:
This was copied verbatim from package/hub/schema/index.js and I'm not sure about the reasons it was done that way then. I'll call this Answer No. 1 so that I can refer to it in my other comments :)